### PR TITLE
add Gemini API sample

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Overall another really nice sample! Left a couple of quick comments, otherwise it looks good.

File: functional-samples/ai.gemini-in-the-cloud/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-in-the-cloud/package.json:1]: Not a fan of committing node_modules (even if it makes things easier). Treeshaking is a good question, usually I'd do it, but in this case ommitted it as it's a really small dep. Might make sense though to always do it by default...

[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/ai.gemini-in-the-cloud/package.json:1]: +1 to just always using a bundler or build step. if there are any deps at all

File: functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    console.error(e);
+    console.log('Prompt:', prompt);
+    throw e;
+  }
+}
+
+async function initDefaults() {
+  generationConfig = initModel();
+  console.log('Model default:', generationConfig);
+  sliderTemperature.value = generationConfig.temperature;
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js]: Seems like temperature default is not provided for the 1.5 flash model. Changed to manually setting the default.

File: functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  hide(elementLoading);
+  show(elementResponse);
+  // Make sure to preserve line breaks in the response
+  elementResponse.textContent = '';
+  const paragraphs = response.split(/\r?\n/);
+  for (const paragraph of paragraphs) {
+    if (paragraph) {
+      elementResponse.appendChild(document.createTextNode(paragraph));
+    }
+    elementResponse.appendChild(document.createElement('BR'));
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js]: Then we wouldn't get paragraphs as Gemini formats them using empty lines (use "Write a poem" prompt as an example). Instead the correct thing is to not add a BR for the final paragraph.

File: functional-samples/ai.gemini-in-the-cloud/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-in-the-cloud/package.json:1]: Added a bundle step

File: functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      category: HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT,
+      threshold: HarmBlockThreshold.BLOCK_NONE
+    }
+  ];
+  genAI = new GoogleGenerativeAI(apiKey);
+  model = genAI.getGenerativeModel({
+    model: 'gemini-1.5-flash',
+    safetySettings,
+    generationConfig: {
+      // maxOutputTokens: 1000,
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js]: Removed it instead...I don't think explaining the params is the job of this sample.

**Review by sebastianbenz — COMMENTED** [review]
Addressed the comments. PTAL

**Review by oliverdunk — APPROVED** [review]
LGTM - thanks for taking the time to address the feedback :)

File: functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliverdunk) [reply functional-samples/ai.gemini-in-the-cloud/sidepanel/index.js]: Ah, that sounds good.

**Pull request merged**